File: app/src/main/java/com/cbruegg/mensaupb/adapter/DishViewModelAdapter.java
package com.cbruegg.mensaupb.adapter;

import android.databinding.DataBindingUtil;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.cbruegg.mensaupb.R;
import com.cbruegg.mensaupb.databinding.RowDishBinding;
import com.cbruegg.mensaupb.viewmodel.DishViewModel;

/**
 * Adapter responsible for displaying DishViewModels in a RecyclerView.
 */
public class DishViewModelAdapter extends ObservableListAdapter<DishViewModel, DishViewModelAdapter.DishBindingViewHolder> {
    @Override public void onBindViewHolder(@NonNull DishBindingViewHolder holder, DishViewModel item, @NonNull View.OnClickListener onClickListener) {
        holder.binding.setDishViewModel(item);
        holder.binding.setOnClickListener(onClickListener);
    }

    @NonNull @Override public DishBindingViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType, @NonNull LayoutInflater inflater) {
        return new DishBindingViewHolder(inflater.inflate(R.layout.row_dish, parent, false));
    }

    static class DishBindingViewHolder extends RecyclerView.ViewHolder {

        public final RowDishBinding binding;

        public DishBindingViewHolder(View itemView) {
            super(itemView);
            binding = DataBindingUtil.bind(itemView);
        }
    }
}

File: app/src/main/kotlin/com/cbruegg/mensaupb/adapter/DishViewModelAdapter.kt
package com.cbruegg.mensaupb.adapter

import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.cbruegg.mensaupb.R
import com.cbruegg.mensaupb.databinding.RowDishBinding
import com.cbruegg.mensaupb.viewmodel.DishViewModel

/**
 * Adapter responsible for displaying DishViewModels in a RecyclerView.
 */
class DishViewModelAdapter : ObservableListAdapter<DishViewModel, DishViewModelAdapter.DishBindingViewHolder>() {
    override fun onBindViewHolder(holder: DishBindingViewHolder, item: DishViewModel, onClickListener: View.OnClickListener) {
        holder.binding.dishViewModel = item
        holder.binding.onClickListener = onClickListener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int, inflater: LayoutInflater): DishBindingViewHolder {
        return DishBindingViewHolder(inflater.inflate(R.layout.row_dish, parent, false))
    }

    class DishBindingViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val binding: RowDishBinding = DataBindingUtil.bind(itemView)
    }
}
